src/hound_writer.rs

```diff
@@ -0,0 +1,201 @@
+
+use std::fs::File;
+use std::io::BufWriter;
+
+
+/// A monitor wrapper for the hound WavWriter that writes out binauralized
+/// audio
+pub struct HoundWriter {
+    writer: Option<WavWriter<BufWriter<File>>>,
+}
+
+impl HoundWriter {
+    /// Instantiates a new HoundWriter, which wraps the hound WavWriter
+    fn new(file: &'static str, wave_spec: WavSpec) -> Self {
+        let writer = WavWriter::create(file, wave_spec).unwrap();
+
+        Self {
+            writer: Some(writer),
+        }
+    }
+}
+
+impl Component for HoundWriter {
+    type InData = (Vec<f32>, Vec<f32>);
+    type OutData = Result<(), HoundError>;
+
+    /// Appends binauralized audio data to the specified output WAV file
+    fn convert(self: &mut Self, input: (Vec<f32>, Vec<f32>)) -> Result<(), HoundError> {
+        let (left_samps, right_samps) = input;
+        let mut writer = self.writer.take().unwrap();
+
+        // interleave the two streams and write the samples to the WAV file
+        for (left, right) in std::iter::zip(left_samps, right_samps) {
+            writer.write_sample(left).unwrap();
+            writer.write_sample(right).unwrap();
+        }
+
+        // flush after each write to save state of the WAV file in the header
+        let result = writer.flush();
+        self.writer = Some(writer);
+        return result;
+    }
+
+    /// Clean up WavWriter after writing all audio data from pipeline. This
+    /// This happens automatically when the WavWriter is dropped, but
+    /// calling this gives us controlled error checking.
+    fn finalize(self: &mut Self) -> Result<(), ComponentError> {
+        let writer = self.writer.take().unwrap();
+
+        match writer.finalize() {
+            hound::Result::Ok(()) => return Ok(()),
+            hound::Result::Err(hound_error) => {
+                return Err(ComponentError::HoundError(hound_error));
+            }
+        }
+    }
+}
+
+impl ToString for HoundWriter {
+    /// Converts the HoundWriter to a String, i.e. returns its name
+    fn to_string(&self) -> String {
+        "HoundWriter".to_string()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::component::run_component;
+    use hound::{SampleFormat, WavReader};
+
+    use std::f32::consts::PI;
+    use std::fs::remove_file;
+    use std::{io::BufReader, sync::mpsc::channel};
+
+    const SAMP_RATE: u32 = 44100;
+    const BITS_PER_SAMPLE: u16 = 32;
+    const FRAME_SIZE: i32 = 128;
+
+    const C: f32 = 261.61;
+
+    fn create_sine_wave(frames: i32, note: f32) -> Vec<f32> {
+        (0..frames)
+            .map(|x| (x % FRAME_SIZE) as f32 / FRAME_SIZE as f32)
+            .map(|t| (t * note * 2.0 * PI).sin() * (i16::MAX as f32))
+            .collect()
+    }
+
+    // Write 100 sine wav frames to an output file using a WavWriter, and
+    // read it back properly using a WavReader
+    #[test]
+    fn test_wav_writer_reader() {
+        let file_name = "/Users/Skylar.Gilfeather/CyberGrape/sine.wav";
```

Comment:
```suggestion
        let file = NamedTempFile::new().unwrap();
        let file_name = file.path();
```

Reply:
This should make it pass the CI by using a dynamic temporary file instead of a hard-coded path.

Reply:
This is a life saver, thank you!